### ISSUE TITLE
NullPointerException while playing track

### ISSUE THREAD

**Issue by duncte123** [sedmelluq/lavaplayer#203]
On version 1.3.19 the following exception happens a lot during playback of a track.

In total, I have 108 of these events at the time of writing.
https://sentry.io/share/issue/a6a4b53f00a04e868f21d927107882fa/
https://sentry.io/share/issue/ec818b78bbda41e4b3e3b38a3f2fada4/
https://sentry.io/share/issue/60400e67e5f94cae8162105021361687/

**Comment by aikaterna** [comment]
Disclaimer: I am a Lavalink user, although I am using Lavaplayer 1.3.19. My node is hosted in the US-East region.

https://www.youtube.com/watch?v=vwbWNv0SCcA

Expected response generated by LP:
Video unavailable
This video contains content from Magnolia, who has blocked it in your country on copyright grounds.

NPE generated instead:
```
com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Something broke when playing the track.
	at com.sedmelluq.discord.lavaplayer.tools.ExceptionTools.wrapUnfriendlyExceptions(ExceptionTools.java:43) ~[lavaplayer-1.3.19.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.execute(LocalAudioTrackExecutor.java:102) ~[lavaplayer-1.3.19.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.lambda$executeTrack$1(DefaultAudioPlayerManager.java:349) [lavaplayer-1.3.19.jar!/:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_212]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_212]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_212]
Caused by: java.lang.NullPointerException: null
	at com.sedmelluq.discord.lavaplayer.tools.DataFormatTools.extractBetween(DataFormatTools.java:30) ~[lavaplayer-1.3.19.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioTrack.loadTrackFormatsFromFormatStreamMap(YoutubeAudioTrack.java:169) ~[lavaplayer-1.3.19.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioTrack.loadTrackFormats(YoutubeAudioTrack.java:135) ~[lavaplayer-1.3.19.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioTrack.loadBestFormatWithUrl(YoutubeAudioTrack.java:98) ~[lavaplayer-1.3.19.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioTrack.process(YoutubeAudioTrack.java:62) ~[lavaplayer-1.3.19.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.execute(LocalAudioTrackExecutor.java:90) ~[lavaplayer-1.3.19.jar!/:na]
	... 4 common frames omitted
```

**Comment by AGuyNamedJens** [comment]
I actually recieve the same problem but instead i just cant seem to get lavalink to fetch youtube URLs anymore...

`2019-08-08 09:02:44.023 ERROR 10316 --- [ader-2-thread-1] lavalink.server.player.AudioLoader       : Load failed

com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Loading information for a YouTube track failed.
        at com.sedmelluq.discord.lavaplayer.tools.ExceptionTools.wrapUnfriendlyExceptions(ExceptionTools.java:43) ~[lavaplayer-1.3.17.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadTrackWithVideoId(YoutubeAudioSourceManager.java:217) ~[lavaplayer-1.3.17.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadFromUrlWithVideoId(YoutubeAudioSourceManager.java:307) ~[lavaplayer-1.3.17.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadFromMainDomain(YoutubeAudioSourceManager.java:244) ~[lavaplayer-1.3.17.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadNonSearch(YoutubeAudioSourceManager.java:314) ~[lavaplayer-1.3.17.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadItemOnce(YoutubeAudioSourceManager.java:189) ~[lavaplayer-1.3.17.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadItem(YoutubeAudioSourceManager.java:128) ~[lavaplayer-1.3.17.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.checkSourcesForItemOnce(DefaultAudioPlayerManager.java:437) ~[lavaplayer-1.3.17.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.checkSourcesForItem(DefaultAudioPlayerManager.java:419) ~[lavaplayer-1.3.17.jar!/:na]
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.lambda$createItemLoader$0(DefaultAudioPlayerManager.java:218) ~[lavaplayer-1.3.17.jar!/:na]
        at java.util.concurrent.FutureTask.run(java.base@9-internal/FutureTask.java:266) ~[na:na]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@9-internal/ThreadPoolExecutor.java:1158) ~[na:na]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@9-internal/ThreadPoolExecutor.java:632) ~[na:na]
        at java.lang.Thread.run(java.base@9-internal/Thread.java:804) ~[na:na]
Caused by: java.lang.NullPointerException: null
        at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioSourceManager.loadTrackWithVideoId(YoutubeAudioSourceManager.java:213) ~[lavaplayer-1.3.17.jar!/:na]
        ... 12 common frames omitted`

**Comment by duncte123** [comment]
@AGuyNamedJens update to latest lavalink

**Comment by Devoxin** [comment]
This should be fixed by #205. NPEs were generated due to the following reasons:
- playabilityStatus was not `ok`, but Lavaplayer didn't check this -- it was still checking the old `status` property so Lavaplayer proceeded to scrape for more information which was non-existent due to the aforementioned reason.
- As of 1.3.20, it was expecting a `cipher` property that wasn't present on all videos/formats.

**Comment by SerenModz21** [comment]
when i try to play a song, my bot says `Could not play: Loading information for a YouTube track failed.` and then when i look on intelliJ IDEA, it says this 
![image](https://user-images.githubusercontent.com/41450647/63911163-401ec280-ca21-11e9-8c74-8704b81a7aa1.png)

**Comment by Shadorc** [comment]
This is fixed in the latest lavaplayer version (1.21)

**Comment by Thewsomeguy** [comment]
It appears that this issue has returned after YouTube made some breaking changes.

Related: https://github.com/fent/node-ytdl-core/issues/486

```
com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Something broke when playing the track.
	at com.sedmelluq.discord.lavaplayer.tools.ExceptionTools.wrapUnfriendlyExceptions(ExceptionTools.java:43) ~[lavaplayer-1.3.21.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.execute(LocalAudioTrackExecutor.java:102) ~[lavaplayer-1.3.21.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.lambda$executeTrack$1(DefaultAudioPlayerManager.java:349) ~[lavaplayer-1.3.21.jar!/:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
Caused by: java.lang.RuntimeException: java.lang.NullPointerException
	at com.sedmelluq.discord.lavaplayer.tools.JsonBrowser.as(JsonBrowser.java:114) ~[lavaplayer-1.3.21.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioTrack.loadTrackFormatsFromStreamingData(YoutubeAudioTrack.java:224) ~[lavaplayer-1.3.21.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioTrack.loadTrackFormats(YoutubeAudioTrack.java:137) ~[lavaplayer-1.3.21.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioTrack.loadBestFormatWithUrl(YoutubeAudioTrack.java:99) ~[lavaplayer-1.3.21.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.source.youtube.YoutubeAudioTrack.process(YoutubeAudioTrack.java:63) ~[lavaplayer-1.3.21.jar!/:na]
	at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.execute(LocalAudioTrackExecutor.java:90) ~[lavaplayer-1.3.21.jar!/:na]
	... 4 common frames omitted
Caused by: java.lang.NullPointerException: null
```

**Comment by Frederikam** [comment]
Youtube is not providing a `contentLength` for all formats. I am working on a fix.

**Comment by Fabricio20** [comment]
This issue has been fixed by PR #210 (Mentioned wrong fix for Issue #203).
